src/core/cpu_profiler.cc

```diff
@@ -123,15 +123,31 @@ void cpu_profiler::on_signal() {
         return;
     }
 
+    // During exception handling in libgcc there is a critical section
+    // where the stack is being modified so execution can be returned to
+    // a handler for the exception. This modification isn't capture by 
+    // the eh_frames for the program though. So when libgcc's backtrace
+    // enters the partially modified stack it will follow invalid addresses
+    // and cause a segfault. To avoid this we check if any exception
+    // is currently being unwound and avoid taking a profiling sample if so.
+    //
+    // Note: this only protects against C++ exceptions, therefore foreign
+    // exceptions or long jumps could still cause segfaults within the profiler.
+    const bool no_uncaught_exceptions = std::uncaught_exceptions() == 0;
```

Comment:
Maybe this is a docs issue but they say:

```
2) Detects how many exceptions in the current thread have been thrown or rethrown and not yet entered their matching catch clauses.
```

So once we enter the catch this should return 0? From my understanding the issue is profiling in the catch handler?

Reply:
Right and so whatever `std::uncaught_exceptions` returns is modified in _cxa_throw and _cxa_catch?

Reply:
Yeah, sorry should've made that clearer. `std::uncaught_exceptions` just returns the mentioned thread-local counter.

Reply:
For way too much information see https://refspecs.linuxfoundation.org/abi-eh-1.22.html#cxx-throw